lib/libzpool/uzfs_io.c

```diff
@@ -60,6 +64,40 @@ uzfs_write_data(zvol_state_t *zv, char *buf, uint64_t offset, uint64_t len,
 
 	rl = zfs_range_lock(&zv->zv_range_lock, r_offset, r_len, RL_WRITER);
 
+	/*
+	 * TODO: error handling in case of UZFS_IO_TX_ASSIGN_FAIL
+	 */
+	if (zv->zv_status && ZVOL_STATUS_REBUILDING) {
```

Comment:
Few comments here:
- this need to be DEGRADED state, so that, we add right from beginning of replica
- function became too big and can be reduced by some macro
- how about using a loop here?